vng_api_common/client.py

```diff
@@ -1,17 +1,67 @@
 """
-Interface to get a zds_client object for a given URL.
+Interface to get a client object for a given URL.
 """
 
-from typing import Optional
+import logging
+import time
+from typing import Any, Optional
 
-from django.apps import apps
 from django.conf import settings
 from django.utils.module_loading import import_string
 
-from zds_client import Client
+import jwt
+from ape_pie import APIClient
+from requests import JSONDecodeError, RequestException, Response
 
+JWT_ALG = "HS256"
 
-def get_client(url: str, url_is_api_root=False) -> Optional[Client]:
+
+logger = logging.getLogger(__name__)
+
+
+class ClientError(RuntimeError):
+    pass
+
+
+# TODO: use more approriate method name
+def to_internal_data(response: Response) -> dict | list | None:
+    try:
+        response_json = response.json()
+    except JSONDecodeError:
+        logger.exception("Unable to parse json from response")
+        response_json = None
+
+    try:
+        response.raise_for_status()
+    except RequestException as exc:
+        if response.status_code >= 500:
+            raise
+        raise ClientError(response_json if response_json is not None else {}) from exc
+
+    assert response_json
+    return response_json
+
+
+class Client(APIClient):
```

Comment:
If possible, I would prefer if we use `zgw_consumers.client.ZGWClient` for this, I think it does everything we need: https://github.com/maykinmedia/zgw-consumers/blob/ce82492a8d82703345f057c24359827b69cc1a6a/zgw_consumers/client.py#L91-L114

Reply:
The link points to the `ZGWAuth` class but the `ZGWClient` was [moved](https://github.com/maykinmedia/zgw-consumers/commit/48c8f8ed1cc6e406886a9d775ea3694665df3f28#diff-0607b90d5b0cd2063b12f5c914bb297d1810c90610b1366aac9fff4765d8f029L34) to the `legacy` module and will be [removed](https://zgw-consumers.readthedocs.io/en/latest/changelog.html#id9) in the future.

Reply:
ah you're right, I thought I removed this comment, I was mainly referring to the auth_headers part, but I added another comment for that